openqabot/loader/gitea.py

```diff
@@ -437,16 +437,11 @@ def add_packages_from_patchinfo(
 
 
 def add_packages_from_files(incident: dict[str, Any], token: dict[str, str], files: list[Any], *, dry: bool) -> None:
-    patchinfo_file = next(
-        (
-            file_info
-            for file_info in files
-            if file_info.get("filename", "").split("/")[-1] == "_patchinfo" and file_info.get("raw_url") is not None
-        ),
-        None,
-    )
-    if patchinfo_file:
-        add_packages_from_patchinfo(incident, token, patchinfo_file["raw_url"], dry=dry)
+    for file_info in files:
+        file_name = file_info.get("filename", "").split("/")[-1]
+        raw_url = file_info.get("raw_url")
+        if file_name == "_patchinfo" and raw_url is not None:
+            add_packages_from_patchinfo(incident, token, raw_url, dry=dry)
```

Comment:
I am fine to change that but I would prefer a less procedural and more functional style. You can write it differently but the "for file_info in files:" is for me a symptom that we should improve on. How about splitting the query from the action with a generator expression and processing that

```
patchinfo_urls = (f["raw_url"] for f in files if f.get("filename", "").split("/")[-1] == "_patchinfo" and f.get("raw_url"))
for url in patchinfo_urls:
    add_packages_from_patchinfo(incident, token, url, dry=dry)
```

Reply:
I'm really wondering why you hate good old for-loops so much. You say it isn't your preferred style but I'm wondering what actual problem you're trying to solve here. I'm also wondering why you like these generator expressions so much. They don't increase readability in my opinion (in fact they decrease readability). They can be nice to avoid intermediate arrays but I don't think this applies here.